src/v/cloud_storage/remote_partition.cc

```diff
@@ -754,9 +754,47 @@ ss::future<bool> remote_partition::tolerant_delete_object(
 
 static constexpr ss::lowres_clock::duration erase_timeout = 60s;
 static constexpr ss::lowres_clock::duration erase_backoff = 1s;
+static constexpr ss::lowres_clock::duration erase_non_0th_delay = 200ms;
+
+/**
+ * Return the index of this node in the list of voters, or nullopt if it
+ * is not a voter.
+ */
+std::optional<size_t>
+voter_position(raft::vnode self, const raft::group_configuration& raft_config) {
+    const auto& voters = raft_config.current_config().voters;
+    auto position = std::find(voters.begin(), voters.end(), self);
+    if (position == voters.end()) {
+        return std::nullopt;
+    } else {
+        return position - voters.begin();
+    }
+}
+
+ss::future<remote_partition::finalize_result> remote_partition::finalize(
+  ss::abort_source& as,
+  raft::vnode self,
+  raft::group_configuration raft_config) {
+    vlog(_ctxlog.info, "Finalizing remote storage state...");
+
+    // To reduce redundant re-uploads in the typical case of all replicas
+    // being alive, have all non-0th replicas delay before attempting to
+    // reconcile the manifest. This is just a best-effort thing, it is
+    // still okay for them to step on each other: finalization is best
+    // effort and the worst case outcome is to leave behind a few orphan
+    // objects if writes were ongoing while deletion happened.
+    auto my_position = voter_position(self, raft_config);
```

Comment:
0th voter position is a little unusual of a condition -- curious why opt for this over, say, delaying on non-leaders.

Not to say we should change it, just curious if there are considerations here I'm missing.

Reply:
At the point we're doing deletion, raft already shut down -- there is no leader.

To make all this a bit less fuzzy would require a deeper change to how deletion works, to have a "deleting" state in which a partition is still running at the raft/storage layer, but not visible at the Kafka layer.  Once we do that (perhaps a bit further out), deletion can belong to the leader the same way uploads do.